### PR TITLE
Refactor trusswork modals to use react-modal

### PR DESCRIPTION
**Pull request by emyl3**
# FRONTEND PULL REQUEST

## Related Issue

- Part one to resolve #7569

## Changes Proposed
- Refactor our usage of trussworks' modals to use our implementation of [react-modal](https://www.npmjs.com/package/react-modal)

## Additional Information
- Effort to upgrade our version of [trussworks/react-uswds](https://github.com/trussworks/react-uswds) to v9.0.0 
  - Trussworks uses the focus-trap-react library to manage focus inside their implementation of the <Modal> component.
   - However, there’s an issue with that library when running tests in Jest because Jest uses JSDom under the hood. So [there’s this warning](https://github.com/focus-trap/focus-trap?tab=readme-ov-file#testing-in-jsdom).
   - The Trussworks team recommends doing [something like this](https://github.com/trussworks/react-uswds/blob/144fc435b892d2b066cc6bd547ea546747d0bc39/docs/modal_tests.md#modal-tests).  
      - I did that [here](https://github.com/CDCgov/prime-simplereport/pull/7569/files#diff-157cbc99ab0c1c08434708d4fe6c16be8a4a1b1b7f67f2fe985d8a29f7ce1582R1) and mocked it [here](https://github.com/CDCgov/prime-simplereport/pull/7569/files#diff-bc6abb0836b5c207b4232e7db185278d9323082d639a06742779d251935692f4R19). 
      - The tests no longer show the ERROR: Your focus-trap must have at least one container with at least one tabbable node in it at all times but it [doesn’t look like the modals](https://github.com/CDCgov/prime-simplereport/actions/runs/8836296172/job/24262564021?pr=7569#step:6:3389) are being rendered in the tests with this change.

As a result, we are not using Trussworks' modals.

## Testing
- Deployed on dev2
- Verify the refactored close test modal and incomplete ask on entry modal in the test queue works as intended
- Ensure the styling of the other modals are the same as before

## Screenshots / Demos
**Before**
Screenshot 2024-05-02 at 17 57 09

**After**
Screenshot 2024-05-02 at 17 54 17

## Future work
- [New issue](https://github.com/CDCgov/prime-simplereport/issues/7649) created to leverage our common component modal and clean up our styling

### CODE REVIEW

File: frontend/src/scss/base/_prime-styles.scss [hunk 1]
```diff
@@ -511,29 +511,12 @@ $results-dropdown-spacing: #{units(4)} - #{units(2)} - 22px - #{units(4)}; // he
   z-index: 1000;
 }
 
-.prime-modal-buttons {
```
Comment (emyl3) [inline frontend/src/scss/base/_prime-styles.scss:514]: :warning I don't think we were using these styles anywhere

File: frontend/src/app/testQueue/TestCard/TestCard.test.tsx [hunk 2]
```diff
@@ -601,6 +601,12 @@ describe("TestCard", () => {
         )
       ).not.toBeInTheDocument();
 
+      // select results
```
Comment (emyl3) [inline frontend/src/app/testQueue/TestCard/TestCard.test.tsx:604]: ⚠️ not sure how this test was working before 😅

**Review by fzhao99 — APPROVED** [review]
code looks fine to me but if you could ask Kenny for a design sweep, think that'd be wise to make sure they don't see anything off with the modal refactoring

**Comment by emyl3** [comment]
> code looks fine to me but if you could ask Kenny for a design sweep, think that'd be wise to make sure they don't see anything off with the modal refactoring

Oh good call! I'll ask them now!

**Review by bobbywells52 — APPROVED** [review]
LGTM -- thanks for your super thorough work on this! Second Bob's thought's on getting a sign off from Kenny as well

**Review by kenieh — APPROVED** [review]
lgtm! ty elisa

**Comment by emyl3** [comment]
@kenieh has reviewed it and approved! will merge cc @fzhao99 @bobbywells52

**Pull request merged**